Tribler/Test/test_as_server.py

```diff
@@ -125,6 +125,12 @@ def tearDown(self, annotate=True):
         self.assertFalse(delayed_calls, "The reactor was dirty when tearing down the test")
         self.assertFalse(Session.has_instance(), 'A session instance is still present when tearing down the test')
 
+        # Clean the delayed calls to prevent them cascading into the next test case
+        for delayed_call in reactor.getDelayedCalls():
+            self._logger.warn("Cancelling delayed call: %s", repr(delayed_call))
+            delayed_call.cancel()
+        self._logger.warn("Removed all readers and writers from the reactor: %s", repr(reactor.removeAll()))
```

Comment:
I don't think I'm undertanding your comment: if an assertion fails, it raises a AssertionError exception and the function execution aborts, anything behind it will be skipped.